core/src/test/java/org/elasticsearch/action/count/CountRequestBuilderTests.java

```diff
@@ -62,71 +57,23 @@ public static void closeClient() {
+        assertThat(countRequestBuilder.toString(), equalTo(new SearchSourceBuilder().size(0).minScore(CountRequest.DEFAULT_MIN_SCORE)
+                .terminateAfter(SearchContext.DEFAULT_TERMINATE_AFTER).toString()));
     }
 
     @Test
     public void testQueryBuilderQueryToString() {
         CountRequestBuilder countRequestBuilder = client.prepareCount();
         countRequestBuilder.setQuery(QueryBuilders.matchAllQuery());
-        assertThat(countRequestBuilder.toString(), equalTo(new QuerySourceBuilder().setQuery(QueryBuilders.matchAllQuery()).toString()));
+        assertThat(countRequestBuilder.toString(), equalTo(new SearchSourceBuilder().size(0).minScore(CountRequest.DEFAULT_MIN_SCORE)
+                .terminateAfter(SearchContext.DEFAULT_TERMINATE_AFTER).query(QueryBuilders.matchAllQuery())
+                .toString()));
     }
 
     @Test
     public void testStringQueryToString() {
         CountRequestBuilder countRequestBuilder = client.prepareCount();
-        String query = "{ \"match_all\" : {} }";
-        countRequestBuilder.setQuery(new BytesArray(query));
-        assertThat(countRequestBuilder.toString(), containsString("\"query\":{ \"match_all\" : {} }"));
-    }
-
-    @Test
-    public void testXContentBuilderQueryToString() throws IOException {
-        CountRequestBuilder countRequestBuilder = client.prepareCount();
-        XContentBuilder xContentBuilder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
-        xContentBuilder.startObject();
-        xContentBuilder.startObject("match_all");
-        xContentBuilder.endObject();
-        xContentBuilder.endObject();
-        countRequestBuilder.setQuery(xContentBuilder);
-        assertThat(countRequestBuilder.toString(), equalTo(new QuerySourceBuilder().setQuery(xContentBuilder.bytes()).toString()));
-    }
-
-    @Test
-    public void testStringSourceToString() {
-        CountRequestBuilder countRequestBuilder = client.prepareCount();
-        String query = "{ \"query\": { \"match_all\" : {} } }";
-        countRequestBuilder.setSource(new BytesArray(query));
-        assertThat(countRequestBuilder.toString(), equalTo("{ \"query\": { \"match_all\" : {} } }"));
-    }
-
-    @Test
-    public void testXContentBuilderSourceToString() throws IOException {
-        CountRequestBuilder countRequestBuilder = client.prepareCount();
-        XContentBuilder xContentBuilder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
-        xContentBuilder.startObject();
-        xContentBuilder.startObject("match_all");
-        xContentBuilder.endObject();
-        xContentBuilder.endObject();
-        countRequestBuilder.setSource(xContentBuilder.bytes());
-        assertThat(countRequestBuilder.toString(), equalTo(XContentHelper.convertToJson(xContentBuilder.bytes(), false, true)));
-    }
-
-    @Test
-    public void testThatToStringDoesntWipeSource() {
```

Comment:
this test was weird I know, but it was there to make sure that a certain bug wouldn't come back at any time... the bug was that we modified the state of the object when calling toString. I think I would restore it and make it work if possible with the updated codebase.